Backend/utils/cronJobTrendingArtists.js

```diff
@@ -0,0 +1,115 @@
+const { PrismaClient } = require('@prisma/client');
+const prisma = new PrismaClient();
+const cron = require('node-cron');
+
+async function calculateTrendingArtists() {
+    const now = new Date();
+    const oneDayAgo = new Date(now.setDate(now.getDate() - 1));
+
+    // Clear the existing trending artists
+    await prisma.trendingArtist.deleteMany({});
+
+    // Fetch tagged songs from the past day
+    const recentTags = await prisma.song.findMany({
+        where: {
+            taggedAt: {
+                gte: oneDayAgo
+            }
+        },
+        include: {
+            user: true
+        }
+    });
+
+    // Fetch artist searches from the past day
+    const recentSearches = await prisma.artistSearch.findMany({
+        where: {
+            createdAt: {
+                gte: oneDayAgo
+            }
+        },
+        include: {
+            artist: true
+        }
+    });
+
+    // Calculate momentum for each artist
+    const artistMomentum = {};
+
+    // Add tagged songs momentum
+    for (const tag of recentTags) {
+        const artistName = tag.artist;
+        if (!artistMomentum[artistName]) {
+            artistMomentum[artistName] = {
+                tagCount: 0,
+                popularity: 0,
+                searchCount: 0,
+                artistId: null,
+            };
+        }
+        artistMomentum[artistName].tagCount += 1;
+
+        const artist = await prisma.artist.findFirst({
+            where: { name: artistName }
+        });
+
+        if (artist) {
+            artistMomentum[artistName].popularity = artist.popularity;
+            artistMomentum[artistName].artistId = artist.id;
+        }
+    }
+
+    // Add search count momentum
+    for (const search of recentSearches) {
+        const artistName = search.artist.name;
+        if (!artistMomentum[artistName]) {
+            artistMomentum[artistName] = {
+                tagCount: 0,
```

Comment:
wouldn't this just wipe out the tag count you created above?

Reply:
No, it initializes tagCount only if the artist hasn't been added to artistMomentum previously